### DISCUSSION TITLE
Access `request.body` in `genReqId`

### DISCUSSION THREAD

**Discussion by SimenB** [fastify/fastify#4361] [Q&A]
We get messages from Google PubSub, and the body contains a request ID I'd like to use as request ID in our handler.

`request.body` is not set at the time `genReqId` is called. What's the best way to access it (if it's even possible)?

**Comment by mcollina** [comment]
I don't think it's possible. I would also advise against it because with pubsub systems there can be retries involved and you want separate ids for those.
